opencti-platform/opencti-front/src/private/components/analyses/groupings/GroupingPopover.jsx

```diff
@@ -58,6 +62,21 @@ const GroupingPopover = ({ id }) => {
     handleClose();
   };
   const handleCloseEdit = () => setDisplayEdit(false);
+  const handleOpenEnrichment = () => {
```

Comment:
When I see a lot of duplicate code like this we are going to put in a lot of places I'm always I bit stressed : what does Landry or Staff about that ?

Reply:
I'll have a try on the first one and push it directly to the PR to have an idea

Reply:
In fact something very similar to `useDeletion` hook but no context specific

Reply:
> I'll have a try on the first one and push it directly to the PR to have an idea

Bad idea, was improving nothing. And after some thoughts I don't think I find a relevant solution that simplifies stuff. I prefer having those repetition of states rather than a complex generic component that needs to manage every cases. At least here it is simple to understand and tweak if needed.

Reply:
It might be a big refactor but could we make a generic Popover component, logic seems very similar, we could pass EditionContainer component as props, or is there tricky parts that we cannot mutualize?

Reply:
Not a fan of doing it as we will obviously run into edges cases, and start adding a lot of complexity and "if"s in the generic popover. I'm not for over-engineer this thing, but keep things simple and easy to customize for every popover